File: 1-100/21.merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        ret = ListNode(0)
        curNode = ret
        while l1 and l2:
            if l1.val > l2.val:
                curNode.next = l2
                l2 = l2.next
            else:
                curNode.next = l1
                l1 = l1.next
            curNode = curNode.next
        
        curNode.next = l1 if l1 else l2

        return ret.next

File: 1-100/21.merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode ret(0);
        ListNode* pCurNode = &ret;
        while (l1 != nullptr && l2 != nullptr)
        {
            if (l1->val > l2->val)
            {
                pCurNode->next = l2;
                l2 = l2->next;
            }
            else
            {
                pCurNode->next = l1;
                l1 = l1->next;
            }

            pCurNode = pCurNode->next;
        }

        pCurNode->next = l1 == nullptr ? l2 : l1;

        return ret.next;
    }
};